src/tif_model_iterator/tif_kernel_iterator.py

```diff
@@ -267,54 +335,107 @@ def transform_to_polygons(df: pd.DataFrame) -> gpd.GeoDataFrame:
         """
         print("Creating geometry")
         start = timer()
-        # Make squares from the the pixels in order to make connected polygons from them.
-        df["geometry"] = df.apply(
-            lambda x: Polygon(
-                [
-                    (x["rd_x_ul"], x["rd_y_ul"]),
-                    (x["rd_x_ur"], x["rd_y_ur"]),
-                    (x["rd_x_lr"], x["rd_y_lr"]),
-                    (x["rd_x_ll"], x["rd_y_ll"]),
-                    (x["rd_x_ul"], x["rd_y_ul"]),
-                ]
-            ),
-            axis=1,
-        )
 
-        df = df[["geometry", "label"]]
+        if self.square_output:
```

Comment:
Still some columns need to be dropped though

Reply:
Hmmm, fair, but maybe move that to a different function then? As in, if square_output = False, this function name is extremely misleading. (because you are actually not changing anything to polygons)